### PR TITLE
native/periph_can: usability improvements and documentation updates

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR wants to make the `vcan` setup usage for native easier. It does so by:

- 2a5879aecc42c8b1af022bdc2bec11c39e634823: combining setup `README` for virtual can interfaces while adding details on how to connect two can interface instead of relying on `candump`/`cansend` and others.
- 3143a99991bdff0205f8151d8170853288dd4719: add `PERIPH_CAN_FLAGS` to specify the can interface to be used, similar to how `PORT` is used with `tap*`, but using an explicitly distinct variable for this.

I've also removed the `apt-get` instructions for installing the `32bit` libsocketcan instructions since it's not packages in recent distro versions and with #17533 its used through a pkg by default.

### Testing procedure

Follow the `README.native.can.md` instructions to se tup two native can interfaces, and send messages between them as well as from the Linux host.

### Issues/PRs references

~~It depends on #17533 to not have to re-adapt the README.~~

### CODE REVIEW

**Comment by benpicco** [comment]
DOSE doesn't use CAN, but @firas-hamdi has been working with CAN lately

File: boards/native/Makefile.include [hunk 1]
```diff
@@ -101,6 +101,15 @@ ifneq (,$(filter periph_eeprom,$(FEATURES_USED)))
   TERMFLAGS += $(EEPROM_FILE_FLAGS)
 endif
 
+VCAN_IFNUM ?= 0
+VCAN_IFNAME ?= vcan0
+VCAN_IFACE ?= $(VCAN_IFNUM):$(VCAN_IFNAME)
+# set the default vcan interface
+ifneq (,$(filter periph_can,$(FEATURES_USED)))
+  PERIPH_CAN_FLAGS = --can $(VCAN_IFACE)
+  TERMFLAGS += $(PERIPH_CAN_FLAGS)
```
Comment (aabadie) [inline boards/native/Makefile.include:110]: Why not:

```suggestion
  TERMFLAGS += --can $(VCAN_IFACE)
```

?

File: cpu/native/include/can_params.h [hunk 2]
```diff
@@ -30,8 +30,12 @@ extern "C" {
  * @brief Default parameters (device names)
  */
 static const candev_params_t candev_params[] = {
+#if CAN_DLL_NUMOF >= 1
```
Comment (aabadie) [inline cpu/native/include/can_params.h:33]: I don't really understand how the CAN_DLL_NUMOF is supposed to be set and how it integrate with the native can flags (is it possible to have multiple calls to --can ?).
Where is it documented btw ? Maybe we should make it more explicit that it's limited to 2 (and why 2 btw) ?

File: tests/candev/README.native.can.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+```
+
+## Enabling multiple vcan interfaces
+
+By default a single can interface is enabled, a second one can be added
+by setting `CAN_DLL_NUMOF=2`, e.g. `CFLAGS=-DCAN_DLL_NUMOF=2`. These two
+interfaces will not be connected by default but look like if they where on
+different Buses.
+
+Too configure which `vcan` interface each native CAN device should use then
```
Comment (aabadie) [inline tests/candev/README.native.can.md:119]:
```suggestion
To configure which `vcan` interface each native CAN device should use then
```

File: cpu/native/include/can_params.h (hunk 2 above)
Comment (aabadie) [reply cpu/native/include/can_params.h:33]: Thanks, things are clearer now. Just found a typo in the native related README. Please squash!

**Comment by fjmolinas** [comment]
squashed

**Review by aabadie — APPROVED** [review]
ACK

**Comment by fjmolinas** [comment]
Thanks @aabadie!

**Pull request merged**